x/interchainqueries/keeper/grpc_query.go

```diff
@@ -56,3 +59,20 @@ func (k Keeper) QueryResult(goCtx context.Context, request *types.QueryRegistere
 	}
 	return &types.QueryRegisteredQueryResultResponse{Result: result}, nil
 }
+
+func (k Keeper) LastRemoteHeight(goCtx context.Context, request *types.QueryLastRemoteHeight) (*types.QueryLastRemoteHeightResponse, error) {
+	req := contypes.QueryConnectionClientStateRequest{ConnectionId: request.ConnectionId}
+	r, err := k.ibcKeeper.ConnectionClientState(goCtx, &req)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(types.ErrInvalidConnectionID, "Connection not found")
+	}
+	clientState := r.GetIdentifiedClientState().GetClientState()
+
+	m := new(ttypes.ClientState)
+	err = proto.Unmarshal(clientState.Value, m)
+	if err != nil {
+		return nil, sdkerrors.Wrapf(types.ErrProtoUnmarshal, "Can't unmarshal client state")
+	}
+
+	return &types.QueryLastRemoteHeightResponse{Height: m.LatestHeight.RevisionHeight}, err
```

Comment:
The only 'nil' value is acceptable value for the error at this place, could we return nil explicitly?